src/System.Configuration.ConfigurationManager/src/System/Configuration/ImplicitMachineConfigHost.cs

```diff
@@ -23,17 +23,27 @@ internal ImplicitMachineConfigHost(IInternalConfigHost host)
             out string locationConfigPath, IInternalConfigRoot configRoot, params object[] hostInitConfigurationParams)
         {
             // Stash the filemap so we can see if the machine config was explicitly specified
-            _fileMap = (ConfigurationFileMap)hostInitConfigurationParams[0];
+            GetFileMap(hostInitConfigurationParams);
             base.InitForConfiguration(ref locationSubPath, out configPath, out locationConfigPath, configRoot, hostInitConfigurationParams);
         }
 
         public override void Init(IInternalConfigRoot configRoot, params object[] hostInitParams)
         {
             // Stash the filemap so we can see if the machine config was explicitly specified
-            _fileMap = (ConfigurationFileMap)hostInitParams[0];
+            GetFileMap(hostInitParams);
             base.Init(configRoot, hostInitParams);
         }
 
+        private void GetFileMap(object[] parameters)
+        {
+            foreach (object parameter in parameters)
```

Comment:
Is it possible parameter could be null and we should early out on that, too?

Reply:
We don't check in the various host implementations so I didn't go that far.